src/test/unit/services/pathfinder/util.hpp

```diff
@@ -0,0 +1,472 @@
+#ifndef TEST_UNIT_SERVICES_PATHFINDER_UTIL_HPP
+#define TEST_UNIT_SERVICES_PATHFINDER_UTIL_HPP
+
+#include <stan/callbacks/interrupt.hpp>
+#include <stan/callbacks/stream_writer.hpp>
+#include <iostream>
+
+namespace stan {
+namespace test {
+
+struct mock_callback : public stan::callbacks::interrupt {
+  int n;
+  mock_callback() : n(0) {}
+
+  void operator()() { n++; }
+};
+
+class loggy : public stan::callbacks::logger {
+  std::ostream& log_;
+
+ public:
+  loggy(std::ostream& x) : log_(x) {}
+  loggy() : log_(std::cout) {}
+  /**
+   * Logs a message with debug log level
+   *
+   * @param[in] message message
+   */
+  virtual void debug(const std::string& message) { log_ << message << "\n"; }
+
+  /**
+   * Logs a message with debug log level.
+   *
+   * @param[in] message message
+   */
+  virtual void debug(const std::stringstream& message) {
+    log_ << message.str() << "\n";
+  }
+
+  /**
+   * Logs a message with info log level.
+   *
+   * @param[in] message message
+   */
+  virtual void info(const std::string& message) { log_ << message << "\n"; }
+
+  /**
+   * Logs a message with info log level.
+   *
+   * @param[in] message message
+   */
+  virtual void info(const std::stringstream& message) {
+    log_ << message.str() << "\n";
+  }
+
+  /**
+   * Logs a message with warn log level.
+   *
+   * @param[in] message message
+   */
+  virtual void warn(const std::string& message) { log_ << message << "\n"; }
+
+  /**
+   * Logs a message with warn log level.
+   *
+   * @param[in] message message
+   */
+  virtual void warn(const std::stringstream& message) {
+    log_ << message.str() << "\n";
+  }
+
+  /**
+   * Logs an error with error log level.
+   *
+   * @param[in] message message
+   */
+  virtual void error(const std::string& message) { log_ << message << "\n"; }
+
+  /**
+   * Logs an error with error log level.
+   *
+   * @param[in] message message
+   */
+  virtual void error(const std::stringstream& message) {
+    log_ << message.str() << "\n";
+  }
+
+  /**
+   * Logs an error with fatal log level.
+   *
+   * @param[in] message message
+   */
+  virtual void fatal(const std::string& message) { log_ << message << "\n"; }
+
+  /**
+   * Logs an error with fatal log level.
+   *
+   * @param[in] message message
+   */
+  virtual void fatal(const std::stringstream& message) {
+    log_ << message.str() << "\n";
+  }
+};
+
+class values : public stan::callbacks::stream_writer {
```

Comment:
This would probably also be useful higher up, at which point it'd be good to call it something other than `values` given this stuff isn't in the `test` namespace.